src/Compilers/VisualBasic/Portable/CommandLine/VisualBasicCompiler.vb

```diff
@@ -17,13 +15,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         Private Shared s_responseFileName As String
         Private ReadOnly _responseFile As String
         Private ReadOnly _diagnosticFormatter As CommandLineDiagnosticFormatter
+        Private _additionalTextFiles As ImmutableArray(Of AdditionalTextFile)
 
         Protected Sub New(parser As VisualBasicCommandLineParser, responseFile As String, args As String(), clientDirectory As String, baseDirectory As String, sdkDirectory As String, additionalReferenceDirectories As String, analyzerLoader As IAnalyzerAssemblyLoader)
             MyBase.New(parser, responseFile, args, clientDirectory, baseDirectory, sdkDirectory, additionalReferenceDirectories, analyzerLoader)
 
-            _diagnosticFormatter = New CommandLineDiagnosticFormatter(baseDirectory)
+            _diagnosticFormatter = New CommandLineDiagnosticFormatter(baseDirectory, AddressOf GetAdditionalTextFiles)
+            _additionalTextFiles = ImmutableArray(Of AdditionalTextFile).Empty
         End Sub
 
+        Private Function GetAdditionalTextFiles() As ImmutableArray(Of AdditionalTextFile)
```

Comment:
Seems like it is a bug to call this function before ResolveadditionalFilesFromArguments is called.  Can we add a comment for that (or even better an assert)?